packages/api/src/services/erc20-state.ts

```diff
@@ -0,0 +1,40 @@
+import { clients } from "@uma/sdk";
+import { Libs, Json } from "..";
+import { asyncValues } from "../libs/utils";
+type Config = Json;
+export default function (config: Config, libs: Libs) {
+  const { provider, erc20s, collateralAddresses, syntheticAddresses } = libs;
+  async function getTokenState(address: string) {
+    const instance = clients.erc20.connect(address, provider);
+    return asyncValues({
+      address,
+      // just in case these fail, return null
+      decimals: instance.decimals().catch((err) => null),
+      name: instance.name().catch((err) => null),
+      symbol: instance.symbol().catch((err) => null),
+    });
+  }
+  async function updateToken(address: string) {
+    if (await erc20s.has(address)) return;
+    const state = await getTokenState(address);
+    return erc20s.upsert(address, state);
+  }
+  async function updateTokens(addresses: string[]) {
+    return Promise.allSettled(addresses.map(updateToken));
+  }
```

Comment:
Is it the case that this function will not throw, but updateToken will? Are these meant to be private or public functions? As public functions, it could be confusing if these very similar functions have such different error behavior.

Reply:
thats right, this wont throw, its also meant to be internal. I'm adopting the allSettled pattern to resolve all promises even if some happen to throw. The errors do get logged though in the update function to make sure things dont silently fail.  The api for services arent really meant to be used like a library, although they technically could. I will update all calls to clarify which ones are meant to be public though.

Reply:
If these are internal and the error behavior is commented, I think this is totally fine. the main concern was that one method just looked like the plural of the other, but they behave quite differently.

As a side note, I continue to feel like the lack of a distinction (even a superficial one, like a preceding underscore!) between private and public methods can be really confusing because it's unclear what's meant for public consumption. It's one major advantage of typescript (specifically typescript) classes, they have different declared visibilities that require code to make this clear.

Reply:
hmm you are right, its hard to see whats public vs private, but i dont really understand why it matters. If we were just to treat all the functions as public, it doesn't materially affect anything, its just a matter of the developer understanding what the calls do.